File: MvvmSample/app/src/main/java/com/abben/mvvmsample/db/MoviesDataBase.java
package com.abben.mvvmsample.db;

import android.app.Application;

import com.abben.mvvmsample.bean.Movie;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

/**
 * Created by abben on 2017/10/25.
 */
@Database(entities = {Movie.class}, version = 1)
public abstract class MoviesDataBase extends RoomDatabase {
    private static MoviesDataBase sInstance;

    public static MoviesDataBase getsInstance(Application application) {
        if (sInstance == null) {
            synchronized (MoviesDataBase.class) {
                sInstance = Room.databaseBuilder(application, MoviesDataBase.class, "movies.db").build();
            }
        }
        return sInstance;
    }

    public abstract MoviesDao moviesDao();
}

File: MvvmSampleJetpack/app/src/main/java/com/abben/mvvmsamplejetpack/db/MoviesDataBase.kt
package com.abben.mvvmsamplejetpack.db

import android.app.Application
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.abben.mvvmsamplejetpack.bean.Movie

/**
 *  @author abben
 *  on 2019/1/18
 */
@Database(entities = [Movie::class], version = 1)
abstract class MoviesDataBase : RoomDatabase() {

    companion object {
        private var sInstance: MoviesDataBase? = null

        fun getsInstance(application: Application): MoviesDataBase {
            if (sInstance == null) {
                sInstance = Room.databaseBuilder(application, MoviesDataBase::class.java, "movies.db").build()
            }
            return sInstance!!
        }
    }

    abstract fun moviesDao(): MoviesDao
}